File: myplugin/src/main/java/com/tu/myplugin/MyPlugin.java
package com.tu.myplugin;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

import com.android.build.gradle.BaseExtension;

/**
 * Created by tuchuantao on 2021/1/22
 */
public class MyPlugin implements Plugin<Project> {

  @Override
  public void apply(Project project) {
    System.out.println(">>>>registerTransform()<<<<");
    project.getExtensions().findByType(BaseExtension.class)
        .registerTransform(new MyTransform());
  }
}

File: TransformForKotlin/myplugin/src/main/java/com/tu/myplugin/MyPlugin.kt
package com.tu.myplugin

import com.android.build.gradle.BaseExtension

import org.gradle.api.Plugin
import org.gradle.api.Project


/**
 * Created by tuchuantao on 2021/1/22
 */
class MyPlugin : Plugin<Project> {

    override fun apply(project: Project) {
        println(">>>>registerTransform()<<<<")
        project.extensions.findByType(BaseExtension::class.java)
            ?.registerTransform(MyTransform())
    }
}